### ISSUE TITLE
`SetsRequiredMembersAttribute` accepting list of members 

### ISSUE THREAD

**Issue by gurustron** [dotnet/roslyn#65358]
## Background and Motivation

With addition of new C# 11 keyword `required` to mark required members we can use `SetsRequiredMembers` to completely disable the analysis, which leads to several potential problems/inconveniences:

1. Can't describe that ctor initializes only several members:

```csharp
public record Foo 
{
  public required string A { get; init; }
  public required string B { get; init; }
}

public record Bar : Foo
{
   public Bar()
   {
     A = "TEST";
   }
}

var bar = new Bar
{
  B = "ANOTHER TEST" // error here despite `A` initialized 
}
``` 

2. Adding member which is not set in ctor does not produce error:

```csharp
public record Foo 
{
  public required string A { get; init; }
  public required string B { get; init; }
 // public required string C { get; init; }  // - uncommenting will not result in error
}

public record Bar : Foo
{
   [SetsRequiredMembers]
   public Bar(string b)
   {
     A = "TEST";
     B = b;      
   }
}
```
[sharplab demo](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQBECWAhgOYB2A9gM4Au+AxpdgMLnACmAgqYQDYCelfJQDcyKAGZMMdE3QBvZOiWZJUACzoAsgAoAlPMXKjAN0IAndACNz6ALzpSbAO7oAQue0AiT7tFIjAL7IQUhikmZsdORmwOgAYuTk6MgK/iroEQCOAK74EbFYGBzy6MRs1MLo+KT4FeghShIZbDl5bAVwGK4lZXXVtZUNAPRD6Vm5+ZidMj3llf11AUoj6AC06NmkUQC222yktKTE6E74PDwO5NTNlNk819XobGZm0cHIYc1RMW42IPGJFKGdAAbQAyuVKAAlFoTdqaNjbSzPSgAXWBTXcZm0hSsumBqSM6GK9k8ABUAKJgsmePxE7r2SyVInAkIhZBAA=)
## Proposed API

Add ctor to `SetsRequiredMembers` which accepts names of the initialized fields

```diff
namespace Microsoft.CodeAnalysis.Operations
{
namespace System.Diagnostics.CodeAnalysis
{
    /// <summary>
    /// Specifies that this constructor sets all required members for the current type, and callers
    /// do not need to set any required members themselves.
    /// </summary>
    [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = false, Inherited = false)]
#if SYSTEM_PRIVATE_CORELIB
    public
#else
    internal
#endif
    sealed class SetsRequiredMembersAttribute : Attribute
    { 
        public SetsRequiredMembersAttribute(){...}
        public SetsRequiredMembersAttribute(params string[] inits){...}
    }
}
```

## Usage Examples

``` C#
public record Bar : Foo
{
   [SetsRequiredMembers(nameof(A))]
   public Bar()
   {
     A = "TEST";
   }
}
```

## Alternative Designs

- New/Different set of attributes?
- Analyze ctors with compiler

**Comment by 333fred** [comment]
Language features need to be requested on dotnet/csharplang, not roslyn. This is also a duplicate of https://github.com/dotnet/csharplang/discussions/6567, https://github.com/dotnet/csharplang/discussions/6536, and https://github.com/dotnet/csharplang/discussions/6637.
